src/vehicle/lanechange.cpp

```diff
@@ -142,28 +139,28 @@ namespace CityFlow{
         if (vehicle->engine->getCurrentTime() - lastChangeTime < coolingTime) return;
         signalSend = std::make_shared<Signal>();
         signalSend->source = vehicle;
-        auto r = (*rnd)();
         if (vehicle->getCurDrivable()->isLane()) {
             Lane * curLane = (Lane *)vehicle->getCurDrivable();
 
             if (curLane->getLength() - vehicle->getDistance() < 30) return;
             double curEst = vehicle->getGap();
             double outerEst = 0;
-            double expectedGap = vehicle->getLen() + interval * vehicle->getSpeed();
-            Router &router = vehicle->controllerInfo.router;
+            double expectedGap = 2 * vehicle->getLen() + 4 * interval * vehicle->getMaxSpeed();
+            if (vehicle->getGap() > expectedGap || vehicle->getGap() < 1.5 * vehicle->getLen()) return;
 
+            Router &router = vehicle->controllerInfo.router;
             if (curLane->getLaneIndex() < curLane->getBelongRoad()->getLanes().size() - 1){
                 if (router.onLastRoad() || router.getNextDrivable(curLane->getOuterLane())) {
                     outerEst = estimateGap(curLane->getOuterLane());
-                    if (outerEst > curEst + expectedGap)
+                    if (outerEst > curEst + vehicle->getLen())
```

Comment:
Please add necessary comments when submitting a PR if you want to change the car following model or other rules. (Adding the comments on Github rather than in the code). Later we need to add detailed comments about the rules we used in the code.